test/healthcheck/framework/healthcheck.go

```diff
@@ -0,0 +1,382 @@
+	ExecutionCount.Inc()
+	hcStartTime := time.Now()
+	glog.V(4).Infof("checking for %v", upsbrokername)
+	err := WaitForEndpoint(kubeClientSet, "ups-broker", "ups-broker-ups-broker")
+	if err != nil {
+		e = fmt.Errorf("no broker endpoint: %v", err)
+		glog.Error(e)
+		return e
+	}
+
+	url := "http://" + upsbrokername + "." + "ups-broker" + ".svc.cluster.local"
+	broker := &v1beta1.ClusterServiceBroker{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: upsbrokername,
+		},
+		Spec: v1beta1.ClusterServiceBrokerSpec{
+			CommonServiceBrokerSpec: v1beta1.CommonServiceBrokerSpec{
+				URL: url,
+			},
+		},
+	}
+
+	err = util.WaitForBrokerCondition(serviceCatalogClientSet.ServicecatalogV1beta1(),
+		broker.Name,
+		v1beta1.ServiceBrokerCondition{
+			Type:   v1beta1.ServiceBrokerConditionReady,
+			Status: v1beta1.ConditionTrue,
+		},
+	)
+	if err != nil {
+		e = fmt.Errorf("broker not ready: %v", err)
+		glog.Error(e)
+		return e
+	}
+
+	err = util.WaitForClusterServiceClassToExist(serviceCatalogClientSet.ServicecatalogV1beta1(), serviceclassID)
+	if err != nil {
+		e = fmt.Errorf("service class not found: %v", err)
+		glog.Error(e)
+		return e
+	}
+
+	namespace, err = CreateKubeNamespace(kubeClientSet)
+	if err != nil {
+		e = fmt.Errorf("error creating namespace: %v", err)
+		glog.Error(e)
+		return e
+	}
+
+	glog.V(4).Info("Creating a ServiceInstance")
+	instance := &v1beta1.ServiceInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      instanceName,
+			Namespace: namespace.Name,
+		},
+		Spec: v1beta1.ServiceInstanceSpec{
+			PlanReference: v1beta1.PlanReference{
+				ClusterServiceClassExternalName: serviceclassName,
+				ClusterServicePlanExternalName:  "default",
+			},
+		},
+	}
+	operationStartTime := time.Now()
+	instance, err = serviceCatalogClientSet.ServicecatalogV1beta1().ServiceInstances(namespace.Name).Create(instance)
+	if err != nil {
+		e = fmt.Errorf("error creating instance: %v", err)
+		glog.Error(e)
+		return e
+	}
+
+	if instance == nil {
+		e = fmt.Errorf("error creating instance - instance is null")
+		glog.Error(e)
+		return e
+	}
+
+	glog.V(4).Info("Waiting for ServiceInstance to be ready")
+	err = util.WaitForInstanceCondition(serviceCatalogClientSet.ServicecatalogV1beta1(),
+		namespace.Name,
+		instanceName,
+		v1beta1.ServiceInstanceCondition{
+			Type:   v1beta1.ServiceInstanceConditionReady,
+			Status: v1beta1.ConditionTrue,
+		},
+	)
+	if err != nil {
+		e = fmt.Errorf("instance not ready: %v", err)
+		glog.Error(e)
+		return e
+	}
+	ReportOperationCompleted("create_instance", operationStartTime)
+
+	glog.V(4).Info("Verify references are resolved")
+	sc, err := serviceCatalogClientSet.ServicecatalogV1beta1().ServiceInstances(namespace.Name).Get(instanceName, metav1.GetOptions{})
+	if err != nil {
+		e = fmt.Errorf("error getting instance: %v", err)
+		return e
+	}
+
+	if sc.Spec.ClusterServiceClassRef == nil {
+		err = errors.New("ClusterServiceClassRef should not be null")
+		glog.Error(err)
+		return err
```

Comment:
Personally, I'd introduce a function that creates, logs and returns the error. That would reduce these three lines to a single one. If used throughout the file, it would not only shave off a few dozen lines of code, but also make the rest of the code stand out more.

Maybe even introduce something like `assertNonNil()` to really clean up things.